tests/test_formatters.py

```diff
@@ -199,3 +201,128 @@ def test_row_formatter_smoke(sample_size: int = 10):
             raise e
         if idx % 10 == 0:
             print(f"Completed {idx}/{sample_size}")
+
+
+def test_str_diff():
+    old = """Removed line
+Single equal line
+Multiple equal lines
+    equal as well
+Replaced line A""".splitlines()
+
+    new = """Single equal line
+Added line
+Multiple equal lines
+    equal as well
+Replace line B""".splitlines()
+
+    old_diff, new_diff = tabulardelta.formatters.detailed_text_formatter._str_diff(
+        old, new
+    )
+
+    assert (
+        "\n".join(old_diff)
+        == """[diff]
+-Removed line
+ Single equal line
+
+[...]
+-Replaced line A"""
+    )
+
+    assert (
+        "\n".join(new_diff)
+        == """[diff]
+
+ Single equal line
++Added line
+[...]
++Replace line B"""
+    )
+
+
+def example_json(age: int, timestasmp: str) -> str:
+    return """{
+  "users": {
+    "id": 1,
+    "name": "Alice Johnson",
+    "email": "alice.johnson@example.com",
+    "age": <AGE>,
+    "isActive": true,
+    "roles": ["admin", "editor"],
+    "profile": {
+      "bio": "Data scientist with 5 years of experience in machine learning.",
+      "location": "New York",
+      "social": {
+        "linkedin": "linkedin.com/in/alicejohnson",
+        "github": "github.com/alicej"
+      }
+    }
+  },
+  "meta": {
+    "count": 3,
+    "timestamp": "<TIMESTAMP>"
+  }
+}""".replace("<AGE>", str(age)).replace("<TIMESTAMP>", timestasmp)
+
+
+def test_some_value_change_diff():
+    first = pd.DataFrame({"col": [example_json(29, "2025-07-27T10:00:00Z")]})
+    second = pd.DataFrame({"col": [example_json(30, "2025-07-28T02:05:00Z")]})
+    delta = PandasComparator().compare(first, second)
+    assert (
+        DetailedTextFormatter().format(delta)
```

Comment:
This is maybe a personal preference (so feel free to ignore if you feel strongly in the other way), but I would prefer these string outputs to be defined elsewhere (e.g., as a constant at the top of the file, some function elsewhere or a pytest fixture). I find it hard to read the code doing the testing when there are these multiline string objects all over. I get the other side of the argument that the test output is right by the code, but as I said, it is just my preference.